test/utils.py

```diff
@@ -376,6 +424,11 @@ def compare_value(bundle1, bundle2, v1, v2, exclude=None):
     elif isinstance(v1, list) and isinstance(v2, list):
         for vv1, vv2 in zip(v1, v2):
             compare_value(bundle1, bundle2, vv1, vv2, exclude)
+    elif isinstance(v1, set) and isinstance(v2, set) and len(v1) == len(v2) and len(v1) <= 1:
```

Comment:
Am I going nuts or are the `bundle1` and `bundle2` arguments in `compare_dictionaries` and `compare_value` not doing anything whatsoever? If true and they do not do anything, do we really need this custom dictionary compare function? Surely there must be existing tools that provide this? I realize this is not something you added in this PR, but just wondering if I am missing something

Reply:
I think you are right, bundle1 and bundle2 are useless! https://github.com/seperman/deepdiff may be an option, but I don't think I will add it in this PR